upgrades/upgrade_testing_framework/core/docker_framework_client.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import docker.client
+from docker.errors import ImageNotFound, NotFound
+from docker.models.networks import Network
+from docker.models.containers import Container
+from docker.models.volumes import Volume
+
+
+class RemoveRunningContainerException(Exception):
+    def __init__(self, container: Container):
+        self.container = container
+        super().__init__("A running container should be stopped before attempting to remove: {}".format(container.id))
+
+
+class ExistingContainerException(Exception):
+    def __init__(self, container_name: str):
+        self.container_name = container_name
+        super().__init__("There already exists a container with the name: {}".format(container_name))
+
+
+class DockerFrameworkClient:
+
+    def __init__(self, logger=logging.getLogger(__name__), docker_client=docker.from_env()):
+        # TODO: Use a custom logger
+        self.logger = logger
+        self.docker_client = docker_client
+
+    def is_container_running(self, container: Container) -> bool:
+        # TODO: Make more robust, case where this attr is not valid
+        if container is None:
+            raise TypeError("Provided argument is not of type docker.models.containers.Container")
```

Comment:
I think I agree with your comment, this becomes an ever expanding cost to always check for None especially when we don't have good reason to think None will be given. I generally am of the mindset to try to think through all possible inputs, but this is a bit overkill

It may be, I saw it referenced to use 'State' attr somewhere but haven't gotten to play with it fully in docker. If you want to use status it should be as simple as:
`return container.status == "running"`

I'm not sure which of these attributes are more reliable for this, but some playing with them should show

Reply:
Hmm, good point.  While `getattr` is safe when you might be invoking it on a `None`, we're saying that we're not expecting that here so we could just use the `status` member.